azkaban-common/src/test/java/azkaban/executor/ExecutionLogsDaoTest.java

```diff
@@ -109,4 +109,32 @@ public void testLargeUploadLog() throws ExecutorManagerException {
     assertThat(logsResult3).isNotNull();
     assertThat(logsResult3.getLength()).isEqualTo(185493);
   }
+
+  @Test
+  public void testLogCleanup() throws ExecutorManagerException {
+    final File logDir = ExecutionsTestUtil.getFlowDir(LOG_TEST_DIR_NAME);
+
+    // Multiple of 255 for Henry the Eigth
+    final File[] largelog =
+        {new File(logDir, "largeLog1.log"), new File(logDir, "largeLog2.log"),
+            new File(logDir, "largeLog3.log")};
+
+    this.executionLogsDao.uploadLogFile(1, "largeFiles", 0, largelog);
+
+    final LogData logsResult = this.executionLogsDao.fetchLogs(1, "largeFiles", 0, 0, 64000);
+    assertThat(logsResult).isNotNull();
+    assertThat(logsResult.getLength()).isEqualTo(64000);
+
+    final LogData logsResult2 = this.executionLogsDao.fetchLogs(1, "largeFiles", 0, 1000, 64000);
+    assertThat(logsResult2).isNotNull();
+    assertThat(logsResult2.getLength()).isEqualTo(64000);
+
+    final LogData logsResult3 = this.executionLogsDao.fetchLogs(1, "largeFiles", 0, 150000, 250000);
+    assertThat(logsResult3).isNotNull();
+    assertThat(logsResult3.getLength()).isEqualTo(185493);
+
+    final long currentTimeMillis = System.currentTimeMillis();
+    int totalRemovedRecords = executionLogsDao.removeExecutionLogsByTime(currentTimeMillis, 3);
+    assertThat(totalRemovedRecords).isEqualTo(7);
```

Comment:
Clarification: How are we asserting that it should be 7? couldn't get the earlier test lines.

Reply:
The size of logFile 2 is very small. It makes only one entry in execution_logs table. The size of logFile 1 makes 3 chunks in database. I have updated test case to simplify it.